integration/sawtooth_integration/tests/test_dynamic_network.py

```diff
@@ -0,0 +1,244 @@
+            'rounds': 1,
+            'start_nodes_per_round': 3,
+            'stop_nodes_per_round': 0,
+            'batches': 5,
+            'time_between_batches': 1,
+        })
+
+###
+
+    def _test_dynamic_network(
+            self,
+            processors=NodeController.intkey_config_registry,
+            peering=NodeController.peer_to_preceding_only,
+            rounds=2,
+            start_nodes_per_round=3,
+            stop_nodes_per_round=1,
+            batches=10,
+            time_between_batches=1,
+            poet_kwargs=None):
+        '''
+        The tests that actually run are just wrappers around
+        this function. The intent is to make it easy to tweak
+        test parameters without having to poke around in the
+        test logic.
+
+        In each round,
+            1) some nodes are started,
+            2) transactions are sent around in various ways,
+            3) block consensus is checked, and
+            4) some nodes are (possibly) stopped.
+        '''
+
+        poet_kwargs = {} if poet_kwargs is None else poet_kwargs
+
+        for round_ in range(rounds):
+            self.start_new_nodes(
+                processors, peering,
+                start_nodes_per_round, round_, poet_kwargs)
+            self.send_populate_batch(time_between_batches)
+            self.send_txns_alternating(batches, time_between_batches)
+            self.send_txns_all_at_once(batches, time_between_batches)
+            self.assert_consensus()
+            self.stop_nodes(stop_nodes_per_round)
+
+###
+
+    def send_txns_all_at_once(self, batches, time_between_batches):
+        for client in self.clients.values():
+            for batch_num in range(batches):
+                self.send_increment_batch(
+                    client, batch_num, time_between_batches)
+
+    def send_txns_alternating(self, batches, time_between_batches):
+        for batch_num in range(batches):
+            for client in self.clients.values():
+                self.send_increment_batch(
+                    client, batch_num, time_between_batches)
+
+    def send_increment_batch(self, client, batch_num, time_between_batches):
+        LOGGER.info('Sending batch {} @ {}'.format(batch_num, client.url))
+        client.send_txns(self.increment)
+        time.sleep(time_between_batches)
+
+    def send_populate_batch(self, time_between_batches):
+        LOGGER.info('Sending populate txns')
+        self.earliest_client().send_txns(self.populate)
+        time.sleep(time_between_batches)
+
+    def start_new_nodes(self,
+                        processors,
+                        peering,
+                        start_nodes_per_round,
+                        round_,
+                        poet_kwargs):
+        start = round_ * start_nodes_per_round
+        for num in range(start, start + start_nodes_per_round):
+            self.start_node(num, processors, peering, poet_kwargs)
+        time.sleep(CATCH_UP_TIME)
```

Comment:
Is there a way that we can actually know if the node has caught up (via the rest api, perhaps)?